docs/proposals/hpa/federated-hpa.md

```diff
@@ -0,0 +1,135 @@
+-->
+
+### Goals
+1. Bring HPA from single cluster to multiple clusters.
+1. Keep the HPA related CRDs of multi-cluster version the same as the single cluster version.
+1. Tolerate the disaster of member cluster or karmada control plane.
+1. It is better to integrate well with the scenarios such as workloads shifting and cloud burst.
+1. It is better to support different implementations of HPA.
+
+### Non-Goals
+1. Deal the workloads with different subset of services.
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+For a platform developer using Kubernetes, I have implemented CD system depending on the single cluster version of HPA. Now I want to use Karmada to run apps on multiclusters. Because the Karmada version of FederatedHPA CRDs are the same as the single version, so I don't need to pay efforts to modify the code of ecosystem. HPA CRs created for the single cluster version still work for the Karmada version of FederatedHPA.
+
+#### Story 2
+For an application developer, I create a HPA CR for the application running on Karmada with FederatedHPA enabled.
+```
+target cpu util 30%
+min replica 3
+max replica 100
+```
+Suddenly, one of the member clusters which my application running on stops to work and can't scale up new pods. Unfortunately, a request burst is comming into the application. The CPU util of pods becomes higher than 30%. It will need 100 Pods totally to take the request burst. Thanks to the good design of the Karmada FederatedHPA, the FederatedHPAController detects failure and modify the HPA CR in another healthy cluster to increase the `max` value. Finally, new pods are scaled on another healthy member cluster successfully. 
+
+#### Story 3
+As an administrator of the Karmada&Kubernetes platform, I receive an alert that the Karmada control plane stops to work and any requests to the Karmada control plane are failed. There are many applications running on the platform heavilly depend on the HPA to handle the unpredictable burst of requests. The chance of RCA occurred becomes really high if the system can't tolerate the failure of federation control plane.
+Thanks to the design of Karmada FederatedHPA, the single cluster version of HPA is reused and still works as expected in the member cluster without the help of Karmada control plane or any other federated controller.
```

Comment:
Maybe the description is not very accurate, but it is really needed.

Thanks to the design of the FederatedHPA, member clusters HPA capacity should not be affected when fed control plane breaks down, because each member cluster's HPAController can work without fed control plane.

Reply:
@kerthcet Thanks for your suggestion. I've corrected in the last commit.